File: src/TextFormatterContext.ts
import { TextFormatterStrategy } from './TextFormatterStrategy';

export class TextFormatterContext {
  private strategy: TextFormatterStrategy;

  constructor(strategy: TextFormatterStrategy) {
    this.strategy = strategy;
  }

  setStrategy(strategy: TextFormatterStrategy): void {
    this.strategy = strategy;
  }

  format(text: string): string {
    return this.strategy.format(text);
  }
}

File: src/main/java/TextFormatterContext.java
public class TextFormatterContext {
    private TextFormatterStrategy strategy;

    public TextFormatterContext(TextFormatterStrategy strategy) {
        this.strategy = strategy;
    }

    public void setStrategy(TextFormatterStrategy strategy) {
        this.strategy = strategy;
    }

    public String format(String text) {
        return strategy.format(text);
    }
}
